### ISSUE TITLE
Negative offset breaks PostgreSQL when relation size is zero.

### ISSUE THREAD

**Issue by taiki-t** [rmosolgo/graphql-ruby#895]
This happening to me:  https://github.com/rmosolgo/graphql-ruby/issues/802

I could fix this with: https://github.com/taiki-t/graphql-ruby/commit/eb11cf1a396c4878dcfab81108a94078816c064f

but I'm not sure how to add a test case for this; the case where connections have no data or `StarWars::Base`s in existing test cases.

It'd be great if I could get guidance to add such test cases or actual test cases for this.

**Comment by esjee** [comment]
Hi! Could you post the query that is causing the issue? It might be helpful to know whether you're using `first` or `last`, or perhaps some more exotic features.

**Comment by taiki-t** [comment]
Hi!
Here is the fragment, entire query is too big and the connection is this part:

```
fragment Comments_post on Post {
  id
  ...PostCard_post
  comments(last: $count, before: $cursor) {
    edges {
      node {
        id
        ...Comment_comment
        __typename
      }
      cursor
    }
    pageInfo {
      startCursor
      hasPreviousPage
      endCursor
      hasNextPage
    }
  }
}
```

Where `$count` is any number and cursor is `nil`
I'm fetching some comments from last ordered by created time ascending.
This works well when comments exist but fails when there are no comments.

https://github.com/rmosolgo/graphql-ruby/blob/a5f4e3bcc545060b1de06fdc70467c7b76792d48/lib/graphql/relay/relation_connection.rb#L71

In my case when there are no comments `relation_count(items)` calculates to `0` and `last` is `2` so it becomes negative offset. `relation_offset(items)` is `nil`.

**Comment by Willianvdv** [comment]
Fixed in #907
